File: trunk/bin/prepare-srf2pepxml.pl
#!/usr/bin/perl -w
use Cwd 'abs_path';
use strict;
use warnings;

my $file_db = abs_path('../DB/YEAST.ensembl50.aa');
my $file_script = "run-srf2pepxml.sh";
my $dir_output = "sequest.pepxml";

open(SCRIPT,">$file_script");
print SCRIPT "#!/bin/bash\n";
foreach my $file_srf (`ls SRF/*.srf`) {
  chomp($file_srf);
  my $file_old_pepxml = $file_srf;
  $file_old_pepxml =~ s/SRF/sequest\.pepxml/;
  $file_old_pepxml =~ s/srf$/xml/g;
  my $file_new_pepxml = $file_old_pepxml;
  $file_new_pepxml =~ s/xml$/sequest\.pepxml/;
  print STDERR $file_srf,"\n";
  print SCRIPT 'echo "process ',$file_srf,"\"\n";
  print SCRIPT join(" ", 'bioworks_to_pepxml.rb',
                  '--dbpath', $file_db,
                  '--outdir', $dir_output, $file_srf),"\n";
  print SCRIPT "mv $file_old_pepxml $file_new_pepxml\n";
}
close(SCRIPT);
`chmod 744 $file_script`;

#671* bioworks_to_pepxml.rb --dbpath ../DB/YEAST.ensembl50.aa --outdir sequest.pepxml/ SRF/062404_005a.srf 

File: pipeline/prepare-srf2pepxml.py
#!/usr/bin/python
import os
import sys
import stat
import massspec_toolbox_config as conf

(db_name, filename_fasta) = conf.get_dbinfo()
filename_script = 'run-srf2pepxml.sh'
dirname_output = 'sequest.pepxml'

f_script = open(filename_script,'w')
f_script.write("#!/bin/bash\n")

for filename_srf in os.listdir('./SRF'):
    filename_srf = filename_srf.strip()
    if( not filename_srf.endswith('.srf') ):
        continue
    filename_old_pepxml = os.path.join(dirname_output,filename_srf.replace('.srf','')+'.xml')
    filename_new_pepxml = os.path.join(dirname_output,filename_old_pepxml.replace('.xml','')+'.sequest.pepxml')
    sys.stderr.write("Process %s ... \n"%filename_srf)
    f_script.write('echo "processing %s ..."\n'%filename_srf)
    f_script.write('bioworks_to_pepxml.rb --dbpath %s --outdir %s %s\n'%(filename_fasta,dirname_output,os.path.join('SRF',filename_srf)))
    f_script.write('mv %s %s\n'%(filename_old_pepxml,filename_new_pepxml))
f_script.close()
os.chmod(filename_script,stat.S_IRWXU)
